### ISSUE TITLE
Fail to copy 1TB file from B2 to S3

### ISSUE THREAD

**Issue by tom77g** [ncw/rclone#2848]
#### What is the problem you are having with rclone?
Error copying or syncing from B2 to S3

#### What is your rclone version (output from `rclone version`)
rclone v1.45
- os/arch: windows/amd64
- go version: go1.11

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Win 10 64 bit

####  Which cloud storage system are you using? (eg Google Drive)
Backblaze B2 to AWS S3

#### The command you were trying to run (eg `rclone copy /tmp remote:tmp`)
rclone sync -vv b2:VeeamOffsiteBackup\CBB_Veeam\F:\B2D\B2Cloud\B2Cloud-1-CritTest\B2Cloud-1-CritTestD2018-12-11T113029.vbk:\20181212034601\B2Cloud-1-CritTestD2018-12-11T113029.vbk aws:critbucket

#### A log from the command with the `-vv` flag (eg output from `rclone -vv copy /tmp remote:tmp`)

```
C:\Rclone>rclone sync -vv b2:VeeamOffsiteBackup\CBB_Veeam\F:\B2D\B2Cloud\B2Cloud-1-CritTest\B2Cloud-1-CritTestD2018-12-11T113029.vbk:\20181212034601\B2Cloud-1-CritTestD2018-12-11T113029.vbk aws:critbucket
2018/12/19 00:26:23 DEBUG : rclone: Version "v1.45" starting with parameters ["rclone" "sync" "-vv" "b2:VeeamOffsiteBackup\\CBB_Veeam\\F:\\B2D\\B2Cloud\\B2Cloud-1-CritTest\\B2Cloud-1-CritTestD2018-12-11T113029.vbk:\\20181212034601\\B2Cloud-1-CritTestD2018-12-11T113029.vbk" "aws:critbucket"]
2018/12/19 00:26:23 DEBUG : Using config file from "C:\\Users\\tom77\\.config\\rclone\\rclone.conf"
2018/12/19 00:26:24 DEBUG : pacer: Reducing sleep to 0s
2018/12/19 00:26:24 INFO  : S3 bucket critbucket: Waiting for checks to finish
2018/12/19 00:26:24 INFO  : S3 bucket critbucket: Waiting for transfers to finish
2018/12/19 00:26:25 DEBUG : B2Cloud-1-CritTestD2018-12-11T113029.vbk: Reading sha1 from header - "none"
2018/12/19 00:26:25 DEBUG : B2Cloud-1-CritTestD2018-12-11T113029.vbk: Reading sha1 from info - ""
2018/12/19 00:27:24 INFO  :
Transferred:      451.996M / 1018.719 GBytes, 0%, 7.489 MBytes/s, ETA 38h40m36s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:      1m0.3s
Transferring:
 *      B2Cloud-1-CritTestD2018-12-11T113029.vbk:  0% /1018.719G, 8.653M/s, 33h28m24s

2018/12/19 00:28:24 INFO  :
Transferred:          525M / 1018.719 GBytes, 0%, 4.362 MBytes/s, ETA 66h23m46s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:      2m0.3s
Transferring:
 *      B2Cloud-1-CritTestD2018-12-11T113029.vbk:  0% /1018.719G, 329.321k/s, 900h33m40s

2018/12/19 00:29:02 ERROR : B2Cloud-1-CritTestD2018-12-11T113029.vbk: Failed to copy: MultipartUpload: upload multipart failed
        upload id: lBif9At_P4lGY23EuaiM9ACt1ZuA6fa2WEHN7PNEAHafsNbcN8PUV0mayO1oG04aaGluwf9IF1eZdVAeMJhEFeuK5S5MSwwiTb_ptOWEceXrS.hD6vOitucRkTGtl25x
caused by: ReadRequestBody: read multipart upload data failed
caused by: unexpected EOF
2018/12/19 00:29:02 ERROR : S3 bucket critbucket: not deleting files as there were IO errors
2018/12/19 00:29:02 ERROR : S3 bucket critbucket: not deleting directories as there were IO errors
2018/12/19 00:29:02 ERROR : Attempt 1/3 failed with 1 errors and: MultipartUpload: upload multipart failed
        upload id: lBif9At_P4lGY23EuaiM9ACt1ZuA6fa2WEHN7PNEAHafsNbcN8PUV0mayO1oG04aaGluwf9IF1eZdVAeMJhEFeuK5S5MSwwiTb_ptOWEceXrS.hD6vOitucRkTGtl25x
caused by: ReadRequestBody: read multipart upload data failed
caused by: unexpected EOF
2018/12/19 00:29:02 INFO  : S3 bucket critbucket: Waiting for checks to finish
2018/12/19 00:29:02 INFO  : S3 bucket critbucket: Waiting for transfers to finish
2018/12/19 00:29:03 DEBUG : B2Cloud-1-CritTestD2018-12-11T113029.vbk: Reading sha1 from header - "none"
2018/12/19 00:29:03 DEBUG : B2Cloud-1-CritTestD2018-12-11T113029.vbk: Reading sha1 from info - ""
2018/12/19 00:29:24 INFO  :
Transferred:      719.884M / 1019.260 GBytes, 0%, 3.991 MBytes/s, ETA 72h35m7s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:      3m0.3s
Transferring:
 *      B2Cloud-1-CritTestD2018-12-11T113029.vbk:  0% /1018.719G, 8.153M/s, 35h32m1s

2018/12/19 00:30:24 INFO  :
Transferred:        1.054G / 1019.260 GBytes, 0%, 4.489 MBytes/s, ETA 64h31m21s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:      4m0.3s
Transferring:
 *      B2Cloud-1-CritTestD2018-12-11T113029.vbk:  0% /1018.719G, 2.674M/s, 108h19m4s

2018/12/19 00:31:24 INFO  :
Transferred:        1.089G / 1019.260 GBytes, 0%, 3.712 MBytes/s, ETA 78h0m42s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:      5m0.3s
Transferring:
 *      B2Cloud-1-CritTestD2018-12-11T113029.vbk:  0% /1018.719G, 619.077k/s, 479h2m30s

2018/12/19 00:31:30 ERROR : B2Cloud-1-CritTestD2018-12-11T113029.vbk: Failed to copy: MultipartUpload: upload multipart failed
        upload id: XrXFHw2P75uDP3x8RKWFlGCiZlB8czTyjrvSas0DW9Hzmnuouht..HwV5YwGCC3K2nt7fx4MYiQxi_eDFTCoDfgzKjij7mpebIot0k45ChQqa_4z1fflU6BimWYpVxBR
caused by: ReadRequestBody: read multipart upload data failed
caused by: unexpected EOF
2018/12/19 00:31:30 ERROR : S3 bucket critbucket: not deleting files as there were IO errors
2018/12/19 00:31:30 ERROR : S3 bucket critbucket: not deleting directories as there were IO errors
2018/12/19 00:31:30 ERROR : Attempt 2/3 failed with 1 errors and: MultipartUpload: upload multipart failed
        upload id: XrXFHw2P75uDP3x8RKWFlGCiZlB8czTyjrvSas0DW9Hzmnuouht..HwV5YwGCC3K2nt7fx4MYiQxi_eDFTCoDfgzKjij7mpebIot0k45ChQqa_4z1fflU6BimWYpVxBR
caused by: ReadRequestBody: read multipart upload data failed
caused by: unexpected EOF
2018/12/19 00:31:30 INFO  : S3 bucket critbucket: Waiting for checks to finish
2018/12/19 00:31:30 INFO  : S3 bucket critbucket: Waiting for transfers to finish
2018/12/19 00:31:31 DEBUG : B2Cloud-1-CritTestD2018-12-11T113029.vbk: Reading sha1 from header - "none"
2018/12/19 00:31:31 DEBUG : B2Cloud-1-CritTestD2018-12-11T113029.vbk: Reading sha1 from info - ""
2018/12/19 00:32:24 INFO  :
Transferred:        1.590G / 1019.808 GBytes, 0%, 4.518 MBytes/s, ETA 64h6m24s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:      6m0.3s
Transferring:
 *      B2Cloud-1-CritTestD2018-12-11T113029.vbk:  0% /1018.719G, 9.595M/s, 30h11m4s

2018/12/19 00:33:24 INFO  :
Transferred:        1.628G / 1019.808 GBytes, 0%, 3.966 MBytes/s, ETA 73h1m16s
Errors:                 0
Checks:                 0 / 0, -
Transferred:            0 / 1, 0%
Elapsed time:      7m0.3s
Transferring:
 *      B2Cloud-1-CritTestD2018-12-11T113029.vbk:  0% /1018.719G, 1.729M/s, 167h30m46s

2018/12/19 00:34:01 ERROR : B2Cloud-1-CritTestD2018-12-11T113029.vbk: Failed to copy: MultipartUpload: upload multipart failed
        upload id: 8uMrB1pJjZ0Tp22mDOHjUUVY9UpG57Jk8utIOD2mfG1d4SMU47oh8CKMgNktoELOhmuAHB7wPfpkJYNos.scqNNF85RVn3Q_aTRmt70ob22nRudZJ2GfKwd3bPJtSPWu
caused by: ReadRequestBody: read multipart upload data failed
caused by: unexpected EOF
2018/12/19 00:34:01 ERROR : S3 bucket critbucket: not deleting files as there were IO errors
2018/12/19 00:34:01 ERROR : S3 bucket critbucket: not deleting directories as there were IO errors
2018/12/19 00:34:01 ERROR : Attempt 3/3 failed with 1 errors and: MultipartUpload: upload multipart failed
        upload id: 8uMrB1pJjZ0Tp22mDOHjUUVY9UpG57Jk8utIOD2mfG1d4SMU47oh8CKMgNktoELOhmuAHB7wPfpkJYNos.scqNNF85RVn3Q_aTRmt70ob22nRudZJ2GfKwd3bPJtSPWu
caused by: ReadRequestBody: read multipart upload data failed
caused by: unexpected EOF
2018/12/19 00:34:01 Failed to sync: MultipartUpload: upload multipart failed
        upload id: 8uMrB1pJjZ0Tp22mDOHjUUVY9UpG57Jk8utIOD2mfG1d4SMU47oh8CKMgNktoELOhmuAHB7wPfpkJYNos.scqNNF85RVn3Q_aTRmt70ob22nRudZJ2GfKwd3bPJtSPWu
caused by: ReadRequestBody: read multipart upload data failed
caused by: unexpected EOF

C:\Rclone>
```

**Comment by ncw** [comment]
Hmm, the underlying error there is `unexpected EOF` which is usually caused by the source stream closing unexpectedly too early.  So what it looks like is that the b2 end stopped delivering the data stream.

This happened at about the same point each time.

Can you download the file from B2 to your local disk using rclone?

**Comment by tom77g** [comment]
Yes, I can download the file from B2 to local disk using rclone:
rclone sync -vv b2:b2-snapshots-d716a595b4bf local:

The issue still occurs when I transfer from B2 to S3.

**Comment by ncw** [comment]
I think this issue would be fixed by https://github.com/ncw/rclone/issues/2108 which is due to be implemented soon!

I suspect that the very long duration transfer from B2 is being timedout somehow.

**Comment by ncw** [comment]
I've merged this to master now which means it will be in [the latest beta](https://beta.rclone.org) in 15-30 mins and released in v1.47
